python/catkin/builder.py

```diff
@@ -287,17 +288,8 @@ def get_multiarch():
 
 def get_python_install_dir():
     # this function returns the same value as the CMake variable PYTHON_INSTALL_DIR from catkin/cmake/python.cmake
```

Comment:
The Python version of the tool calling `cmake` is not relevant for this. CMake will make a choice and use it throughout the process. So whatever is chosen in CMake (which can be influenced by passing `PYTHON_VERSION`) decides the install path for Python code.